### PR TITLE
switched to first-order splines

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

The script used to generate the C1T crosstrack vs. downtrack visualization has been shown to poorly fit splines to the route graph, which causes the reported crosstrack error to be higher than what is actually experienced. This PR includes a fix to improve spline fitment and report more accurate crosstrack error.

## Motivation and Context

Crosstrack error is an important metric to track as we establish the performance capabilities and limitations of the C1T system, so accurately reporting this metric is crucial.

## How Has This Been Tested?

New splines were fit to a route driven in the main lab and visualized. The crosstrack vs. downtrack error was also compared to the route-driven in 2D space and visually better aligns with the deviations experienced by the truck.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by john-chrosniak** [comment]
route_driven
new_crosstrack_error
new_spline_fitment

File: src/carma-1-tenth/plot_crosstrack_error.py [hunk 1]
```diff
@@ -97,6 +98,8 @@ def plot_absolute_route_deviation(bag_dir, start_offset=0.0):
 
     print("Average Deviation:", np.mean(np.abs(route_deviations)))
     print("Maximum Deviation:", np.max(np.abs(route_deviations)))
+    # plt.plot(route_x_points, route_y_points)
```
Comment (willjohnsonk) [inline src/carma-1-tenth/plot_crosstrack_error.py]: If these are left commented out we should identify what they do with another comment.

**Review by willjohnsonk — CHANGES_REQUESTED** [review]
Looks good besides request for added comment

File: src/carma-1-tenth/plot_crosstrack_error.py (hunk 1 above)
Comment (john-chrosniak) [reply src/carma-1-tenth/plot_crosstrack_error.py]: Those are no longer needed, I removed them in my most recent commit

**Review by willjohnsonk — APPROVED** [review]

**Pull request merged**